File: src/ur_script_sender_py/ur_script_sender_py/send_script.py
import time
import rclpy
from rclpy.node import Node
from std_msgs.msg import String

SCRIPT_TOPIC = "/urscript_interface/script_command"

class UrScriptSender(Node):
    def __init__(self):
        super().__init__("ur_script_sender")
        self.pub = self.create_publisher(String, SCRIPT_TOPIC, 10)

    def send_once(self):
        script = (
            'def my_prog():\n'
            '  set_digital_out(1, True)\n'
            '  movej(p[0.2, 0.3, 0.8, 0, 0, 3.14], a=1.2, v=0.25, r=0)\n'
            '  textmsg("motion finished")\n'
            'end\n'
        )
        msg = String()
        msg.data = script
        self.pub.publish(msg)
        self.get_logger().info("URScript published.")

def main():
    rclpy.init()
    node = UrScriptSender()

    # subscriber 매칭 시간 조금 주기 (안전)
    time.sleep(0.2)
    node.send_once()

    # publish가 wire로 나갈 시간 약간
    rclpy.spin_once(node, timeout_sec=0.1)

    node.destroy_node()
    rclpy.shutdown()

if __name__ == "__main__":
    main()

File: src/ur_script_sender_cpp/src/send_script.cpp
#include <rclcpp/rclcpp.hpp>
#include <std_msgs/msg/string.hpp>

static const char* SCRIPT_TOPIC = "/urscript_interface/script_command";

int main(int argc, char** argv)
{
  rclcpp::init(argc, argv);
  auto node = std::make_shared<rclcpp::Node>("ur_script_sender_cpp");
  auto pub = node->create_publisher<std_msgs::msg::String>(SCRIPT_TOPIC, 10);

  std::string script =
    "def my_prog():\n"
    "  set_digital_out(1, True)\n"
    "  movej(p[0.2, 0.3, 0.8, 0, 0, 3.14], a=1.2, v=0.25, r=0)\n"
    "  textmsg(\"motion finished\")\n"
    "end\n";

  std_msgs::msg::String msg;
  msg.data = script;

  // subscriber 매칭 시간을 조금 주는 게 안전
  rclcpp::sleep_for(std::chrono::milliseconds(200));
  pub->publish(msg);
  RCLCPP_INFO(node->get_logger(), "URScript published.");

  rclcpp::shutdown();
  return 0;
}
